persistence/src/main/scala/hmda/persistence/institutions/SubmissionPersistence.scala

```diff
@@ -29,16 +29,23 @@ object SubmissionPersistence {
       event match {
         case SubmissionCreated(s) =>
           SubmissionState(s +: submissions)
+
         case SubmissionStatusUpdated(id, status) =>
-          val x = submissions.find(x => x.id == id).getOrElse(Submission())
+          val x: Submission = submissions.find(x => x.id == id).getOrElse(Submission())
           val i = submissions.indexOf(x)
-          if (status == Signed) {
-            SubmissionState(submissions.updated(i, x.copy(status = status, end = System.currentTimeMillis())))
-          } else {
-            SubmissionState(submissions.updated(i, x.copy(status = status)))
-          }
+
+          val updatedSub: Submission = if (status == Signed) {
```

Comment:
I would rather have the logic after the `=` encapsulated by `{ }` to improve readability a bit